consensus/src/lib.rs

```diff
@@ -253,7 +253,15 @@ impl Consensus {
                 .map(|(_, certificate)| certificate)
                 .collect();
         }
-        parents.contains(&prev_leader)
+
+        // We check if there are f+1 instances of prev_leader in the parents to determine if there are f+1 unique paths
+        let mut count = 0;
+        for p in parents.iter() {
+            if p.eq(&prev_leader) {
+                count = count + 1;
+            }
+        }
+        count >= self.committee.validity_threshold()
```

Comment:
Not sure this is correct if the leaders are not consecutive. You are only checking there are `f+1` links on the last round. Also, it is more 'rusty' to use iterators here rather than for loops.

Reply:
Updated to match the logic of the overleaf code